compiler/src/dmd/sarif.d

```diff
@@ -165,44 +165,16 @@ Params:
   format = A format string for constructing the error message.
   ap = A variable argument list used with the format string.
   kind = The kind of error (error, warning, deprecation, note, or message).
+  executionSuccessful = `true` if execution succeeded; includes an empty `results` array in the SARIF report.
 
 Throws:
   This function is marked as `nothrow` and does not throw exceptions.
 
 See_Also:
   $(LINK2 https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0.json, SARIF 2.1.0 schema)
 */
-void generateSarifReport(const ref SourceLoc loc, const(char)* format, va_list ap, ErrorKind kind) nothrow
+void generateSarifReport(const ref SourceLoc loc, const(char)* format = null, va_list ap = null, ErrorKind kind = ErrorKind.message, bool executionSuccessful = true) nothrow
```

Comment:
why the default parameters? also `executionSuccessful` is `false` for most call site, why default it into true?